plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -108,6 +110,17 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             icebergCatalog = icebergCatalogInstance;
         }
 
-        return new TrinoRestCatalog(icebergCatalog, catalogName, sessionType, trinoVersion, typeManager, uniqueTableLocation);
+        ImmutableMap.Builder<String, String> credentials = ImmutableMap.builder();
```

Comment:
If you want to filter only credential properties. Consider something like:
```
        Map<String, String> credentials = securityProperties.entrySet().stream()
                .filter(entry -> Set.of(TOKEN, CREDENTIAL).contains(entry.getKey())).
                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
```
which may better fit intention